src/shiv/cli.py

```diff
@@ -111,6 +114,11 @@ def _copytree(src: Path, dst: Path) -> None:
     default=False,
     help="Add the contents of the zipapp to PYTHONPATH (for subprocesses).",
 )
+@click.option(
+    "--reproducible-build-id/--no-reproducible-build-id",
+    default=False,
+    help="Generate reproducible build id for zipapp instead of the random one.",
+)
```

Comment:
let's just call this "--reproducible/--not-reproducible" and use it to gate the timestamp nulling
I think we ought to have it implicitly true if SOURCE_DATE_EPOCH is in the env as well. Click has an `envvar` kwarg for options, but I don't think it's necessarily compatible with boolean options (and we need the contents of the env var if supplied).

Reply:
Changed the option name. I think envvar won't work here, but I've added the logic for this option being implicitely true if SOURCE_DATE_EPOCH is set